### DISCUSSION TITLE
Calendar

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#273] [Component suggestions]
<img width="720" alt="visuele weergave van het calendar component" src="https://github.com/nl-design-system/backlog/assets/84851545/7cdfbf43-3e8a-4e3b-b696-9b66ca5c5b72">

We willen graag meer leren over het Calendar component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1718791776309409)

---

## Naam
Calendar

## Doel
Een kalender waarin je een beschikbare datum kan kiezen.

## Gerelateerde componenten
[Date Input](https://github.com/orgs/nl-design-system/discussions/270), [Date Input Group](https://github.com/orgs/nl-design-system/discussions/271), [Date Picker](https://github.com/orgs/nl-design-system/discussions/272)

## Onderzoek

[Calendar component - North Health Design System](https://nordhealth.design/components/calendar/)
Implementatie van de Calendar component bij het North Health Design System.

[Calendar component- Skyscanner Design System](https://www.skyscanner.design/latest/components/calendar/web-DDIY8XIe)
Implementatie van de Calendar component bij het Skyscanner Design System. Bevat default, range en localized varianten.

[Calendar component - NS Design System](https://design.ns.nl/4a05a30ad/p/53a2b7-calendar/b/16f2c2)
Implementatie van de Calendar component bij het NS Design System.

## Notities
Niet veel losse Calendar componenten tegengekomen bij overheidsorganisaties. Wel in combinatie met een Date Picker component. Deze versies worden in de [Discussion voor de Date Picker](https://github.com/orgs/nl-design-system/discussions/272) component vastgelegd.

**Comment by MariekeBrouwer** [comment]
Groningen gebruikt de kalender wanneer het doel van het formulier ook is om een afspraak te maken. Voorbeeld hiervan vanuit qmatic. 
Nog een aanvulling hiervoor: voor dit soort selectie-opties zou je in gevallen ook direct een tijd willen kunnen selecteren. 
![image](https://github.com/nl-design-system/backlog/assets/149144374/4456b5ff-f511-4931-82b6-f27d57bc311d)

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook-react/index.html?path=/docs/react-calendar--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/calendar)

---

### Screenshots / Varianten

**Storybook**
Default
<img width="1025" height="483" alt="Scherm­afbeelding 2026-01-13 om 12 56 34" src="https://github.com/user-attachments/assets/8deee04a-75a8-42b2-b018-ba11bfb5379d" />

Limited Range Calendar
<img width="1030" height="484" alt="Scherm­afbeelding 2026-01-13 om 12 56 43" src="https://github.com/user-attachments/assets/11d249c9-a4a5-47bd-8f9a-b2d3ba2cd83d" />

English Calendar
<img width="1020" height="481" alt="Scherm­afbeelding 2026-01-13 om 12 56 48" src="https://github.com/user-attachments/assets/8081f6d7-d45d-4541-812e-e12c473742a9" />

---

### Notities

- Varianten beschikbaar voor een bepaalde range aan beschikbare data en engelse versie.

**Reply by jeffreylauwers** [reply]
## Redesign Community component Calendar van Utrecht

- [Design](https://www.figma.com/design/fy08SZpZmqx6ljLwvA3Woe/NL-Design-System---Schetsboek?node-id=9326-19186&t=i6aCWYRQ0oihy6ul-4)
- [Design Tokens](https://github.com/nl-design-system/themes/pull/1303)

**Ontwerpkeuzes (design tokens)**

Toen ik het Community-component Calendar van Utrecht in Figma ontwikkelde, miste ik verschillende design tokens om de ontwerpkeuzes die ik voor ogen had goed door te voeren. Zo miste ik bijvoorbeeld design tokens zoals:

- `utrecht.calendar.table.days-item-day.background-color`, om de default state van de 'days-item-day'-button van een achtergrondkleur te voorzien. Hierdoor wordt visueel duidelijker welke dagen wel of niet interactief zijn en welke binnen de maand vallen. Momenteel erft deze de waarde van `utrecht.button.subtle.background-color`, maar in het Start- en Voorbeeldthema is deze transparant.
- `utrecht.calendar.table.column-gap`, `utrecht.calendar.table.row-gap` of iets als `utrecht.calendar.table.border-spacing`, om de ruimte tussen de dagen in de tabel te kunnen beïnvloeden.

Daarnaast worden momenteel veel ontwerpkeuzes overgenomen van Button (Subtle). Na het bekijken van andere implementaties zie ik dat daar vaak een andere stijl wordt toegepast op de 'days-item-day'-button dan op Button (Subtle). Denk aan verschillen in Font-weight, Padding, Border-radius, etc. 

Daarom denk ik dat er aanvullende design tokens nodig zijn specifiek voor de days-item-day-button.

Zie hier het verschil tussen wat ik voor elkaar kon krijgen (links) en wat ik eigenlijk voor ogen had (rechts):

<img width="860" height="516" alt="Twee Calender componenten naast elkaar. Links een ontwerp op basis van wat er nu mogelijk is met design tokens, rechts een ontwerp zoals ik deze graag had opgezet." src="https://github.com/user-attachments/assets/50bc8221-264c-49e9-8cc6-c21dd630ee27" />

**Naamgeving**

Mocht er een redesign komen, dan zou ik ook graag een voorstel doen voor een aanpassing in de naamgeving van sub-componenten:

Calendar
- Header
  - Button
  - Button
  - Label
  - Button
  - Button
- Weekdays
  - Weekday
  - Weekday
  - Weekday
  - etc.
- Days
  - Day
  - Day
  - Day
  - etc.
  
Dit resulteert dan ook in een aanpassing van de naamgeving voor de design tokens:

- `todo.calendar.background-color`
- `todo.calendar.border-color`
- `todo.calendar.border-radius`
- `todo.calendar.min-inline-size`
- `todo.calendar.day.background-color`
- `todo.calendar.day.border-color`
- `todo.calendar.day.border-radius`
- `todo.calendar.day.border-width`
- `todo.calendar.day.color`
- `todo.calendar.day.font-familiy`
- `todo.calendar.day.font-size`
- `todo.calendar.day.font-weight`
- `todo.calendar.day.line-heigth`
- `todo.calendar.day.min-block-size`
- `todo.calendar.day.min-inline-size`
- `todo.calendar.day.padding-block-end`
- `todo.calendar.day.padding-block-start`
- `todo.calendar.day.padding-inline-end`
- `todo.calendar.day.padding-inline-start`
- `todo.calendar.day.active.background-color`
- `todo.calendar.day.active.border-color`
- `todo.calendar.day.active.color`
- `todo.calendar.day.disabled.background-color`
- `todo.calendar.day.disabled.border-color`
- `todo.calendar.day.disabled.color`
- `todo.calendar.day.disabled.font-weight`
- `todo.calendar.day.focus.background-color`
- `todo.calendar.day.focus.border-color`
- `todo.calendar.day.focus.color`
- `todo.calendar.day.hover.background-color`
- `todo.calendar.day.hover.border-color`
- `todo.calendar.day.hover.color`
- `todo.calendar.day.next-month.background-color`
- `todo.calendar.day.next-month.border-color`
- `todo.calendar.day.next-month.color`
- `todo.calendar.day.next-month.font-weight`
- `todo.calendar.day.next-month.active.background-color`
- `todo.calendar.day.next-month.active.border-color`
- `todo.calendar.day.next-month.active.color`
- `todo.calendar.day.next-month.hover.background-color`
- `todo.calendar.day.next-month.hover.border-color`
- `todo.calendar.day.next-month.hover.color`
- `todo.calendar.day.previous-month.background-color`
- `todo.calendar.day.previous-month.border-color`
- `todo.calendar.day.previous-month.color`
- `todo.calendar.day.previous-month.font-weight`
- `todo.calendar.day.previous-month.active.background-color`
- `todo.calendar.day.previous-month.active.border-color`
- `todo.calendar.day.previous-month.active.color`
- `todo.calendar.day.previous-month.hover.background-color`
- `todo.calendar.day.previous-month.hover.border-color`
- `todo.calendar.day.previous-month.hover.color`
- `todo.calendar.day.selected.background-color`
- `todo.calendar.day.selected.border-color`
- `todo.calendar.day.selected.color`
- `todo.calendar.day.selected.font-weight`
- `todo.calendar.day.selected.active.background-color`
- `todo.calendar.day.selected.active.border-color`
- `todo.calendar.day.selected.active.color`
- `todo.calendar.day.selected.hover.background-color`
- `todo.calendar.day.selected.hover.border-color`
- `todo.calendar.day.selected.hover.color`
- `todo.calendar.day.today.background-color`
- `todo.calendar.day.today.border-color`
- `todo.calendar.day.today.color`
- `todo.calendar.day.today.font-weight`
- `todo.calendar.day.today.active.background-color`
- `todo.calendar.day.today.active.border-color`
- `todo.calendar.day.today.active.color`
- `todo.calendar.day.today.hover.background-color`
- `todo.calendar.day.today.hover.border-color`
- `todo.calendar.day.today.hover.color`
- `todo.calendar.days.column-gap`
- `todo.calendar.days.padding-block-end`
- `todo.calendar.days.padding-block-start`
- `todo.calendar.days.padding-inline-end`
- `todo.calendar.days.padding-inline-start`
- `todo.calendar.days.row-gap`
- `todo.calendar.header.column-gap`
- `todo.calendar.header.color`
- `todo.calendar.header.font-familiy`
- `todo.calendar.header.font-size`
- `todo.calendar.header.font-weight`
- `todo.calendar.header.line-height`
- `todo.calendar.header.padding-block-end`
- `todo.calendar.header.padding-block-start`
- `todo.calendar.header.padding-inline-end`
- `todo.calendar.header.padding-inline-start`
- `todo.calendar.header.button.background-color`
- `todo.calendar.header.button.border-color`
- `todo.calendar.header.button.border-radius`
- `todo.calendar.header.button.border-width`
- `todo.calendar.header.button.color`
- `todo.calendar.header.button.min-block-size`
- `todo.calendar.header.button.min-inline-size`
- `todo.calendar.header.button.padding-block-end`
- `todo.calendar.header.button.padding-block-start`
- `todo.calendar.header.button.padding-inline-end`
- `todo.calendar.header.button.padding-inline-start`
- `todo.calendar.header.button.active.background-color`
- `todo.calendar.header.button.active.border-color`
- `todo.calendar.header.button.active.color`
- `todo.calendar.header.button.disabled.background-color`
- `todo.calendar.header.button.disabled.border-color`
- `todo.calendar.header.button.disabled.color`
- `todo.calendar.header.button.focus.background-color`
- `todo.calendar.header.button.focus.border-color`
- `todo.calendar.header.button.focus.color`
- `todo.calendar.header.button.hover.background-color`
- `todo.calendar.header.button.hover.border-color`
- `todo.calendar.header.button.hover.color`
- `todo.calendar.header.button.icon.size`
- `todo.calendar.weekday.color`
- `todo.calendar.weekday.font-familiy`
- `todo.calendar.weekday.font-size`
- `todo.calendar.weekday.font-weight`
- `todo.calendar.weekday.line-height`
- `todo.calendar.weekday.min-inline-size`
- `todo.calendar.weekday.padding-block-end`
- `todo.calendar.weekday.padding-block-start`
- `todo.calendar.weekday.padding-inline-end`
- `todo.calendar.weekday.padding-inline-start`
- `todo.calendar.weekdays.column-gap`
- `todo.calendar.weekdays.padding-block-end`
- `todo.calendar.weekdays.padding-block-start`
- `todo.calendar.weekdays.padding-inline-end`
- `todo.calendar.weekdays.padding-inline-start`

**Comment by jeffreylauwers** [comment]
## Den Haag

[Calendar ontwerp gevonden in Figma](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=3078-14356&t=PfrsWzaif4lbni3a-4) als onderdeel van Date Picker.

---

### Screenshots / Varianten

**Figma**
![Scherm­afbeelding 2024-10-18 om 12 02 03](https://github.com/user-attachments/assets/e42c113f-2eac-4403-a7af-7775e2a11990)

---

### Notities

[Bekijk de Date Picker GitHub Discussion voor meer informatie](https://github.com/orgs/nl-design-system/discussions/272#discussioncomment-10961411)

**Comment by rianrietveld** [comment]
Onder voorbehoud naar de HOF:
Er moet een goede toegankelijke manier zijn om een of meerdere data te kiezen. Als het niet lukt voor deze component,  of het veel te gecompliceerd wordt, dan een alternatief bieden, zoals select/option constructie.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Calendar op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=calendar)

**Comment by jeffreylauwers** [comment]
## Rotterdam

Als onderdeel van Date Picker component

 - [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/rods-date-picker--docs)
 - [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/packages/web-components/date-picker)

---

### Screenshots / Varianten

**Storybook**
<img width="982" height="643" alt="Scherm­afbeelding 2026-01-08 om 14 15 36" src="https://github.com/user-attachments/assets/66085a9c-a71a-49f1-8b65-26fce1854d32" />

---

### Notities

- Onder de naam: Date Picker. Vandaar ook geplaatst in de [Date Picker GitHub Discussions](https://github.com/orgs/nl-design-system/discussions/272#discussioncomment-10961455).
- Desktop en Mobile layout beschikbaar, waarbij de mobile layout niet langer gebruik maakt van een kalender-weergave maar een formulier binnen een overlay.

**Comment by VincentSmedinga** [comment]
## Amsterdam

* [Storybook](https://designsystem.amsterdam/?path=/docs/components-navigation-calendar--docs)
* [GitHub (CSS)](https://github.com/Amsterdam/design-system/blob/main/packages/css/src/components/calendar)
* [GitHub (React)](https://github.com/Amsterdam/design-system/tree/main/packages/react/src/Calendar)

---

### Screenshots / Varianten

#### Storybook

Standaard:
<img width="704" height="496" alt="ams-calendar-1-default" src="https://github.com/user-attachments/assets/ed718e42-47e2-4878-a27b-f156c0905bc1" />

Met dagen zonder links:
<img width="704" height="496" alt="ams-calendar-2-without-links" src="https://github.com/user-attachments/assets/b777f44a-210d-4301-b5d0-6c483533b58d" />

In het Arabisch:
<img width="704" height="520" alt="ams-calendar-3-arabic" src="https://github.com/user-attachments/assets/0e464467-47ae-479c-a78a-9d8a93dce961" />

---

## Notities

- **Maand- en jaarnavigatie** – knoppen met enkele en dubbele chevrons voor vorige/volgende maand/jaar. Custom iconen mogelijk. Datums buiten de maand worden niet getoond.
- **Datums als links** – dit is een navigatiecomponent. Elke datum verwijst via een instelbare URL naar een dagpagina.
- **Datums zonder link** – het is mogelijk om naar wens datums (via een functie) geen link te geven.
- **Huidige dag gemarkeerd** – vandaag krijgt extra nadruk en `aria-current="date"`.
- **Toegankelijk** – rendert als een `nav`-landmark met een instelbare toegankelijke naam. Schermlezers noemen voor elke dag de volledige datum inclusief weekdag. Tabnavigatie: eerst de buttons, dan alle links achtereenvolgens.
- **Lokalisatie** – de `locale` bepaalt de weekdag- en maandnamen (standaard `nl-NL`). Alle talen mogelijk; de week begint op maandag. De navigatieknoppen hebben instelbare labels. Werkt met talen die van rechts naar links schrijven. Custom iconen voor buttons hebben een HTML-attribuut nodig zodat ze spiegelen.
- **Router-integratie** – bruikbaar in single page applications via een `linkComponent` property.

Visueel precies hetzelfde, maar met buttons voor in een formulier: [Date Picker](https://github.com/orgs/nl-design-system/discussions/272#discussioncomment-17565025).
